src/React.Router/HtmlHelperExtensions.cs

```diff
@@ -0,0 +1,167 @@
+using HttpResponse = Microsoft.AspNetCore.Http.HttpResponse;
+using Microsoft.AspNetCore.Html;
+#endif
+
+namespace React.Router
+{
+	/// <summary>
+	/// Render a React StaticRouter Component with context.
+	/// </summary>
+	public static class HtmlHelperExtensions
+	{
+		/// <summary>
+		/// Gets the React environment
+		/// </summary>
+		private static IReactEnvironment Environment
+		{
+			get
+			{
+				try
+				{
+					return ReactEnvironment.Current;
+				}
+				catch (TinyIoCResolutionException ex)
+				{
+					throw new ReactNotInitialisedException(
+#if NET451
+						"ReactJS.NET has not been initialised correctly.",
+#else
+						"ReactJS.NET has not been initialised correctly. Please ensure you have " +
+						"called services.AddReact() and app.UseReact() in your Startup.cs file.",
+#endif
+						ex
+					);
+				}
+			}
+		}
+
+		/// <summary>
+		/// Render a React StaticRouter Component with context object.
+		/// Can optionally be provided with a custom context handler to handle the various status codes.
+		/// 
+		/// </summary>
+		/// <param name="htmlHelper">MVC Razor <see cref="IHtmlHelper"/></param>
+		/// <param name="componentName">Name of React Static Router component. Expose component globally to ReactJS.NET</param>
+		/// <param name="props">Props to initialise the component with</param>
+		/// <param name="path">F.x. from Request.Path. Used by React Static Router to determine context and routing.</param>
+		/// <param name="Response">Used either by contextHandler or internally to modify the Response status code and redirect.</param>
+		/// <param name="contextHandler">Optional custom context handler, can be used instead of providing a Response object</param>
+		/// <param name="htmlTag">HTML tag to wrap the component in. Defaults to &lt;div&gt;</param>
+		/// <param name="containerId">ID to use for the container HTML tag. Defaults to an auto-generated ID</param>
+		/// <param name="clientOnly">Skip rendering server-side and only output client-side initialisation code. Defaults to <c>false</c></param>
+		/// <param name="serverOnly">Skip rendering React specific data-attributes during server side rendering. Defaults to <c>false</c></param>
+		/// <param name="containerClass">HTML class(es) to set on the container tag</param>
+		/// <returns><see cref="IHtmlString"/> containing the rendered markup for provided React Router component</returns>
+		public static IHtmlString ReactRouterWithContext<T>(
+			this IHtmlHelper htmlHelper, 
+			string componentName,
+			T props,
+			string path = null,
+			HttpResponse Response = null,
+			string htmlTag = null,
+			string containerId = null,
+			bool clientOnly = false,
+			bool serverOnly = false,
+			string containerClass = null,
+			Action<HttpResponse, RoutingContext> contextHandler = null
+		)
+		{
+			try
+			{
+				path = path ?? htmlHelper.ViewContext.HttpContext.Request.Path;
+				Response = Response ?? htmlHelper.ViewContext.HttpContext.Response;
```

Comment:
No, this was just intended as a convenience and i was hoping it might help for unit testing?

But I guess it might just be confusing, I'll remove it